demisto_sdk/commands/lint/helpers.py

```diff
@@ -0,0 +1,350 @@
+    "pwsh_test": 0b100000000
+}
+
+# Power shell checks
+PWSH_CHECKS = ["pwsh_analyze", "pwsh_test"]
+PY_CHCEKS = ["flake8", "bandit", "mypy", "vulture", "pytest", "pylint"]
+
+# Line break
+RL = '\n'
+
+
+def test_internet_connection(url: str = 'http://www.google.com/', timeout: int = 4) -> bool:
+    """ Test internet connection
+
+    Args:
+        url(str): URL to test with.
+        timeout(int): timeout in seconds
+
+    Returns:
+        bool: True if internet connection availble, else False.
+    """
+    try:
+        _ = requests.get(url, timeout=timeout)
+        return True
+    except requests.ConnectionError:
+        print("No internet connection available.")
+    return False
+
+
+def build_skipped_exit_code(no_flake8: bool, no_bandit: bool, no_mypy: bool, no_pylint: bool, no_vulture: bool,
+                            no_test: bool, no_pwsh_analyze, no_pwsh_test, docker_engine: bool) -> bool:
+    """
+    no_flake8(bool): Whether to skip flake8.
+    no_bandit(bool): Whether to skip bandit.
+    no_mypy(bool): Whether to skip mypy.
+    no_vulture(bool): Whether to skip vulture
+    no_pylint(bool): Whether to skip pylint.
+    no_test(bool): Whether to skip pytest.
+    docker_engine(bool): docker engine exists.
+    """
+    skipped_code = 0b0
+    if no_flake8:
+        skipped_code |= EXIT_CODES["flake8"]
+    if no_bandit:
+        skipped_code |= EXIT_CODES["bandit"]
+    if no_mypy or not docker_engine:
+        skipped_code |= EXIT_CODES["mypy"]
+    if no_vulture or not docker_engine:
+        skipped_code |= EXIT_CODES["vulture"]
+    if no_pylint or not docker_engine:
+        skipped_code |= EXIT_CODES["pylint"]
+    if no_test or not docker_engine:
+        skipped_code |= EXIT_CODES["pytest"]
+    if no_pwsh_analyze or not docker_engine:
+        skipped_code |= EXIT_CODES["pwsh_analyze"]
+    if no_pwsh_test or not docker_engine:
+        skipped_code |= EXIT_CODES["pwsh_test"]
+
+    return skipped_code
+
+
+def get_test_modules(content_repo: Optional[git.Repo]) -> Dict[Path, bytes]:
+    """ Get required test modules from content repository - {remote}/master
+    1. Tests/demistomock/demistomock.py
+    2. Tests/scripts/dev_envs/pytest/conftest.py
+    3. Scripts/CommonServerPython/CommonServerPython.py
+    4. CommonServerUserPython.py
+
+    Returns:
+        dict: path and file content - see below modules dict
+    """
+    modules = [Path("Tests/demistomock/demistomock.py"),
+               Path("Tests/scripts/dev_envs/pytest/conftest.py"),
+               Path("Packs/Base/Scripts/CommonServerPython/CommonServerPython.py"),
+               Path("Tests/demistomock/demistomock.ps1"),
+               Path("Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1")]
+    modules_content = {}
+    if content_repo:
+        remote = content_repo.remote()
+        for module in modules:
+            modules_content[module] = content_repo.commit(f'{remote.name}/master').tree[
```

Comment:
What are you doing here? Please provide an explanation.

Reply:
I will add an explaination, I am trying to get the file from the local copy of the remote repo <remote_name>/master before downloading directly from GitHub, So it will be possible to work without internet connection.

Reply:
We should just use the file on disk not from `master`. Otherwise it is impossible to test a change to one of these files.

Reply:
I am not chaning the file it exists in the directory, just if it missing, If it exists I skip creating the file there, I handle it, in  the following context manager - https://github.com/demisto/demisto-sdk/blob/lint/demisto_sdk/commands/lint/helpers.py#L164

Currently every time when using the lint it create the file from scratch (demisto-mock.py which is not wanted behavior)

Reply:
@glicht 
Get it from disk